lib/model/folder_sendrecv.go

```diff
@@ -342,10 +349,16 @@ func (f *sendReceiveFolder) processNeeded(dbUpdateChan chan<- dbUpdateJob, copyC
 				// WithNeed, furthermore, the file can simply be of the wrong
 				// type if we haven't yet managed to pull it.
 				if ok && !df.IsDeleted() && !df.IsSymlink() && !df.IsDirectory() && !df.IsInvalid() {
-					fileDeletions[file.Name] = file
+					fileDeletions.Set(file.Name, &file)
 					// Put files into buckets per first hash
 					key := string(df.Blocks[0].Hash)
-					buckets[key] = append(buckets[key], df)
+					v := &protocol.Index{}
+					if ok := buckets.Get(key, v); ok {
+						v.Files = append(v.Files, df)
+					} else {
+						v.Files = []protocol.FileInfo{df}
+					}
+					buckets.Set(key, v)
```

Comment:
This is somewhat a confusing API, as you assume you get a pointer back, so it's not obvious that you have to Set it back, but it makes sense.

Reply:
I get that, but it's a package that writes to disk, so I'd think it should be somewhat clear that you shouldn't rely on pointer semantics.